src/inmanta/agent/agent.py

```diff
@@ -627,20 +629,34 @@ async def repair_action() -> None:
                 )
             )
 
+        async def cleanup_executors_action() -> None:
+            await self.executor_manager.cleanup_inactive_executors(self._executor_retention)
```

Comment:
I would move this into the executor manager itself. I see little reason for the agent to be responsible for the timer of the manager